src/lib/site-server-process.ts

```diff
@@ -142,12 +142,33 @@ export default class SiteServerProcess {
 		}
 
 		return new Promise< void >( ( resolve, reject ) => {
+			const KILL_TIMEOUT = 5000;
+			let isResolved = false;
+
+			const forceKillTimeout = setTimeout( () => {
+				if ( isResolved ) return;
+
+				// Check if process and pid are still valid before attempting force kill
+				if ( process && process.pid ) {
+					require( 'process' ).kill( process.pid, 'SIGKILL' );
+				} else {
+					if ( ! isResolved ) {
+						isResolved = true;
+						resolve();
+					}
+				}
+			}, KILL_TIMEOUT );
+
 			process.once( 'exit', ( code ) => {
-				if ( code !== 0 ) {
-					reject( new Error( `Site server process exited with code ${ code } upon stopping` ) );
-					return;
+				clearTimeout( forceKillTimeout );
+				if ( ! isResolved ) {
+					isResolved = true;
+					if ( code !== 0 ) {
+						reject( new Error( `Site server process exited with code ${ code } upon stopping` ) );
+					} else {
+						resolve();
+					}
 				}
-				resolve();
 			} );
 			process.kill();
```

Comment:
`process.kil()` returns a boolean per the [Electron docs](https://www.electronjs.org/docs/latest/api/utility-process). Is that a useful signal for us maybe..?

Reply:
That's a nice catch!
I've made some adjustments to use the return value from `process.kill()` instead of the TIMEOUT.
Could you take another look, please? 🙏